src/PostgREST/ApiRequest.hs

```diff
@@ -191,11 +191,19 @@ userApiRequest schema req reqBody =
     where lastWord = last $ T.split (=='.') key
 
   headerRange = if singular && method == "GET" then singletonRange 0 else rangeRequested hdrs
-  urlOffsetRange = rangeGeq . fromMaybe (0::Integer) $
-    readBSMaybe =<< join (lookup "offset" qParams)
-  urlRange = restrictRange
-    (readBSMaybe =<< join (lookup "limit" qParams))
-    urlOffsetRange
+  replaceLast x s = T.intercalate "." $ L.init (T.split (=='.') s) ++ [x]
```

Comment:
On the second thought, if for some reason there is a situation where the array is empty and `L.init` blows, isn't int better to have it like that (the logs would show and undefined error) then quietly hiding the error and making it hard to detect the logical error (empty array) in the first place?